### PR TITLE
Checks if all the libraries in setup.py are listed in installation.rst file

### LINKED ISSUE #8088: Auto-generate installation.rst
**Description**

Not all users know that the instalation.rst file should be updated when they add a new library to the setup.py file. I think we should standardize it so that the tables contained in installation.rst are generated automatically.
https://airflow.readthedocs.io/en/latest/installation.html

Are you wondering how to start contributing to this project? Start by reading our [contributor guide](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst)

**Use case / motivation**

N/A

**Related Issues**

N/A

### PR DESCRIPTION
**Pull request by szn**
Checks if all the libraries `EXTRAS_REQUIREMENTS` section in `setup.py` are documented in the `installation.rst` file.

### Notes

Checking if these two files are in sync is tricky cos there is a lot of freedom in how `setup.py` is documented. Entries in `EXTRAS_REQUIREMENTS` can take these forms:

```
'name': name,
'alias': name
```

In the documentation, this can be reflected as:

```
| name       | ``pip install 'apache-airflow[alias]'`|
| alias      | ``pip install 'apache-airflow[name]'` |
| name       | ``pip install 'apache-airflow[name]'` |
```

The algorithm used is not strict:

1. In the first round, it checks if either packaged name from `setup.py` or any of its aliases is present in the second column in the `installation.rst` file.
2. In the second round, it checks if a name listed in the second column in `installation.rst` is listed as a name or alias in the `setup.py` file.

### Warning

There are differences between these two files. I fixed them to be able to commit the changes. But these packages documentation should be updated:

* pagerduty
* plexus
* sentry
* singularity
* tableau
* virtualenv

closes: #8088

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -565,6 +565,8 @@ def write_version(filename: str = os.path.join(*[my_dir, "airflow", "git_version
     "apache.hive": hive,
     "apache.kylin": kylin,
     "apache.pinot": pinot,
+    "apache.presto": presto,
+    "apache.spark": spark,
```
Comment (szn) [inline setup.py:568]: My intention was not to touch the setup.py file. But these changes seemed logical and safe.  Hope I am right.

File: scripts/ci/pre_commit/pre_commit_check_setup_installation.rst.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    output_table = ""
+
+    for extras in sorted(setup_packages.keys()):
+        if not set(setup_packages[extras]).intersection(docs_packages):
+            output_table += "| {:20} | {:^10} | {:^10} |\n".format(extras, "V", "")
+    
+    setup_packages_str = str(setup_packages)
+    for extras in sorted(docs_packages):
+        if not f"'{extras}'" in setup_packages_str:
```
Comment (szn) [inline scripts/ci/pre_commit/pre_commit_check_setup_installation.rst.py]: I am not very proud of this assertion but it works, seems safe, and I was unable to find an efficient way of flattening a multidimensional array of strings.

File: scripts/ci/pre_commit/pre_commit_check_setup_installation.rst.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        return file_to_read.read()
+
+def get_extras_from_setup() -> {str: bool}:
+    """
+    Returns an array EXTRAS_REQUIREMENTS with aliases from setup.py file in format:
+    {'package name': ['alias1', 'alias2], ...}
+    """
+    setup_content = get_file_content(SETUP_PY_FILE)
+    
+    extras_section_regex = re.compile(
```
Comment (szn) [inline scripts/ci/pre_commit/pre_commit_check_setup_installation.rst.py:47]: I was experimenting with `from setup import EXTRAS_REQUIREMENTS`, but this was I wasn't able to detect aliases.

**Comment by potiuk** [comment]
Some static checks are failing :(

**Comment by szn** [comment]
> Some static checks are failing :(

Fixed except the failing check of newly introduced pre-commit-check (as described in **warning** section above).

**Comment by potiuk** [comment]
I am afraid you need to rebase the commit of yours rather than merge :( @szn . This change becomes rather huge now ;)

**Comment by potiuk** [comment]
Screenshot from 2020-11-01 19-28-27

**Comment by szn** [comment]
@potiuk I know I know. I did rebase but the pressed handful 'arrows' icons in VSC and THIS happened :/

**Pull request closed without merging**